Compiler/SimCode/SimCodeUtil.mo

```diff
@@ -437,11 +438,12 @@ algorithm
     end if;
 
     // add residuals vars from DAE creation
-    if Flags.getConfigBool(Flags.DAE_MODE) then
+    if Flags.getConfigEnum(Flags.DAE_MODE)>1 then
       daeModeSP := createDaeModeSparsePattern(bdaeModeVars, bdaeModeEqns, shared, crefToSimVarHT);
       residualVars := rewriteIndex(residualVars, 0);
       algebraicVars := sortSimVarsAndWriteIndex(algebraicVars, crefToSimVarHT);
-      daeModeData := SOME(SimCode.DAEMODEDATA(daeEquations, daeModeSP, residualVars, algebraicVars));
+      daeModeConf := match Flags.getConfigEnum(Flags.DAE_MODE) case 2 then SimCode.ALL_EQUATIONS(); case 3 then SimCode.DYNAMIC_EQUATIONS(); end match;
```

Comment:
I'm open to other solutions, any proposal?
Actually, exactly in this case I think it's okay to handle with enumeration numbers, they aren't that magic,  but below in the c-code they get more magical, I don't like that solution that much, but it works ;).